Call to 'vec_popcntu' is ambiguous

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

Please:

* Read the documentation to test with the latest developer build.
* Check if other person has already created the same issue to avoid duplicates. You can comment on it if there already is an issue.
* Try to be as detailed as possible in your report.
* Report only one problem per created issue.


This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master, 834c99255320fb565259d3e9177edcb590d4a6be
- Operating System / Platform => Ubuntu 18.04, Power 9
- Compiler => clang version 10.0.0 (git@github.com:llvm/llvm-project.git 2f51a43f8c2b117e69a45eae6a7c849e035a793b)

##### Detailed description
On compilation, clang fails to compile due to an ambiguous call to vec_popcntu. Cut down error message follows but I can prepare a longer dump if need be:
```
In file included from .../opencv/modules/core/src/algorithm.cpp:43:
In file included from .../opencv/modules/core/src/precomp.hpp:85:
In file included from .../opencv/modules/core/include/opencv2/core/hal/intrin.hpp:186:
.../opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:841:21: error: call to 'vec_popcntu' is ambiguous     
{ return v_uint64x2(vec_popcntu(a.val)); }        
                    ^~~~~~~~~~~

.../opencv/modules/core/include/opencv2/core/vsx_utils.hpp:363:1: note: candidate function
                                           ^
In file included from .../opencv/modules/core/src/algorithm.cpp:43:
In file included from .../opencv/modules/core/src/precomp.hpp:85:
In file included from .../opencv/modules/core/include/opencv2/core/hal/intrin.hpp:186:
.../opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:843:21: error: call to 'vec_popcntu' is ambiguous
{ return v_uint64x2(vec_popcntu(a.val)); }
```

I've prepared a solution, pull request to follow.

##### Steps to reproduce
 1. Checkout opencv.
 1. Make build directory.
 1. `cmake <path to source> -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang`
 1. `make -j###`
 1. Receive above error message.

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
 or attach as .txt or .zip file
-->

--- a/modules/core/include/opencv2/core/vsx_utils.hpp
+++ b/modules/core/include/opencv2/core/vsx_utils.hpp
@@ -363,10 +363,12 @@ VSX_FINLINE(Tvec) vec_popcntu(const Tvec2& a)  \
 VSX_IMPL_POPCNTU(vec_uchar16, vec_char16, vec_uchar16_c);
 VSX_IMPL_POPCNTU(vec_ushort8, vec_short8, vec_ushort8_c);
 VSX_IMPL_POPCNTU(vec_uint4,   vec_int4,   vec_uint4_c);
+VSX_IMPL_POPCNTU(vec_udword2, vec_dword2, vec_udword2_c);
 // redirect unsigned types
 VSX_REDIRECT_1RG(vec_uchar16, vec_uchar16, vec_popcntu, vec_popcnt)
 VSX_REDIRECT_1RG(vec_ushort8, vec_ushort8, vec_popcntu, vec_popcnt)
 VSX_REDIRECT_1RG(vec_uint4,   vec_uint4,   vec_popcntu, vec_popcnt)
+VSX_REDIRECT_1RG(vec_udword2, vec_udword2, vec_popcntu, vec_popcnt)
 
 // converts between single and double precision
 VSX_REDIRECT_1RG(vec_float4,  vec_double2, vec_cvfo, __builtin_vsx_xvcvdpsp)

